src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -762,10 +763,16 @@ export class DataViewsService {
    * @param skipFetchFields if true, will not fetch fields
    * @returns DataView
    */
-  async create(spec: DataViewSpec, skipFetchFields = false): Promise<DataView> {
+  async create({ id, ...restOfSpec }: DataViewSpec, skipFetchFields = false): Promise<DataView> {
     const shortDotsEnable = await this.config.get(FORMATS_UI_SETTINGS.SHORT_DOTS_ENABLE);
     const metaFields = await this.config.get(META_FIELDS);
 
+    const spec = {
+      id: id ?? `local-${uuid.v4()}`,
```

Comment:
if we are not relying on special form of id we shouldn't prefix it with `local` here.